### PR TITLE
Generate less strict hanami version on Gemfile

### PR DESCRIPTION
**Pull request by davydovanton**
Reference #501

/cc @jodosha

### CODE REVIEW

**Comment by joneslee85** [comment]
+1

File: lib/hanami/generators/application/app/Gemfile.tt [hunk 1]
```diff
@@ -15,7 +15,7 @@ gem 'hanami-mailer',      require: false, github: 'hanami/mailer'
 gem 'hanami-assets',      require: false, github: 'hanami/assets'
 gem 'hanami',                             github: 'hanami/hanami'
 <%- else -%>
-gem 'hanami',       '<%= Hanami::VERSION %>'
+gem 'hanami',       '~> <%= Hanami::VERSION_WITHOUT_TINY %>'
```
Comment (jodosha) [inline lib/hanami/generators/application/app/Gemfile.tt]: @davydovanton We should unify with `hanami-model` definition and use `<%= config[:hanami_version] %>`.
The logic for defining that version should take care of `Hanami::VERSION`, but **NOT** to define a new constant.

Example:

```ruby
version = Hanami::VERSION.scan(/\A\d{1,2}\.\d{1,2}/).first
```

Comment (AlfonsoUceda) [inline lib/hanami/generators/application/app/Gemfile.tt]: @davydovanton could you take care of this? thanks ;)

Comment (davydovanton) [inline lib/hanami/generators/application/app/Gemfile.tt]: @AlfonsoUceda yes, of course :smiley:

**Comment by davydovanton** [comment]
@AlfonsoUceda @jodosha I created `hanami_version` option.

File: lib/hanami/commands/new/abstract.rb [hunk 2]
```diff
@@ -90,6 +90,10 @@ def hanami_model_version
           @hanami_model_version
         end
 
+        def hanami_version
+          Hanami::VERSION.scan(/\A\d{1,2}\.\d{1,2}/).first
```
Comment (jodosha) [inline lib/hanami/commands/new/abstract.rb]: @davydovanton Please have a look at https://github.com/hanami/hanami/pull/503/files#diff-b6e68c896b787eed14d8d92616b64789R18, there is a discrepancy between `hanami` and `hanami-model` versions. I think we should probably let `#hanami_version` to include `~>` as prefix of the version. Eg. `"~> 0.8"`.

In this way the Ruby code is in complete control of the version number and the template is logic-less, as it only interpolates the value.

Comment (davydovanton) [inline lib/hanami/commands/new/abstract.rb]: Sounds reasonably :+1:

**Comment by davydovanton** [comment]
@jodosha I fixed you comment and also I replaced getter `hanami_model_version` method to `attr_reader` construction.

**Comment by jodosha** [comment]
@davydovanton This looks good :+1: Merging it, thank you :+1:

**Comment by davydovanton** [comment]
@jodosha thanks for review :+1:

**Pull request merged**